### ISSUE TITLE
Non-reachable sites/pages are added into the history

### ISSUE THREAD

**Issue by OlafLostViking** [QupZilla/qupzilla#2333]
Instead of appending everything I write into the address bar to the history of QupZilla, I'd like to propose to only add successful request and drop those that failed. This would include DNS errors (host-not-found) and probably 40x (page-not-found), too.

If 5xx errors should be added is more difficult to decide (I'd say no), since there's probably a valid resource behind that request which simply failed temporarily. But the other two will only add unnecessary URLs into the history -- including proposing them to me (and I chose them often enough because I'm hitting <kbd>enter</kbd> too fast ;) )

**Comment by nowrep** [comment]
Errors should not be added to history at all.

**Comment by OlafLostViking** [comment]
_should not_ or _are not_? ;-) So I tested it again and there are two different situations:

I positively have non-existing hosts in the history even though they also never existed before. But they are all before July 201**6**. (I didn't want to delete those entries before posting a bug report.) So it looks like some update already fixed that behaviour.

The second thing are 404 with some rendered error pages. It also happens here on github. Just enter a nonsense path and you get the "octopus-page" added into the history of QupZilla. I verified with chromiums developer tools - github sends a correct 404 code. (Some nice developer tools for QupZilla would come in handy sometimes :) ).
